### PR TITLE
General components and FE structure

### PR DESCRIPTION
**Pull request by MarekBodingerBA**

### CODE REVIEW

**Review by radoslavzeman — APPROVED** [review]
I prefer not to create empty pages, but to create some TODOs list in index page or somewhere.

Very good job in components page! I love DOs and DONTs style here with examples and source link! ❤️

**Review by mpinter — APPROVED** [review]
Great! Loving this as well ❤️ 

A couple questions and perhaps a followup in comments below. 

I don't mind the empty pages tbh. Would maybe prefer if they contained at least a content of "TODO" to be clear when you opened them that they are unfinished and it isn't an error that they displayed empty. But I like that they show an implied structure that is easy to fill in, would not get rid of them.

File: pages/frontend/components.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+export default Example
+```
+
+ * ✅ Add TSDoc if needed.
+ * ✅ Add Figma link if suitable.
+ * ✅ Prefer types over interfaces for props (consistency).
+ * ❌ Don't prefix interfaces with `I`. ([source](https://stackoverflow.com/a/41967120))
+ * ❌ Don't create a `index.ts` groupping the components. In the ES module system, it adds an unnecessary step and behaviorally leads to creating less components (there should be more of them!). ([example](https://github.com/bratislava/kupaliska-starz-fe/blob/master/src/components/index.ts))
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply pages/frontend/components.md:25]: We still need to discuss file structure (unfortunately).

File: pages/frontend/components.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  return <div className={twMerge(containerClasses, className)}>
+    <h1 className={textClassname}></h1>
+  </div>
+}
+
+```
+
+
+## Use controlled state
+If implementing a component that accepts a value from parent and is able to change it use `useControlledState` from `@react-stately/utils`. It also allows you to skip value from parent and provide a default value if needed.
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply pages/frontend/components.md:74]: Added a resource that explains it.

**Pull request merged**